vendor/fbx/ofbx.h

```diff
@@ -0,0 +1,485 @@
+
+        virtual ~Object() {}
+        virtual Type getType() const = 0;
+
+        const IScene& getScene() const;
+        Object*       resolveObjectLink(int idx) const;
+        Object*       resolveObjectLink(Type type, const char* property, int idx) const;
+        Object*       resolveObjectLinkReverse(Type type) const;
+        Object*       getParent() const;
+
+        RotationOrder getRotationOrder() const;
+        Vec3          getRotationOffset() const;
+        Vec3          getRotationPivot() const;
+        Vec3          getPostRotation() const;
+        Vec3          getScalingOffset() const;
+        Vec3          getScalingPivot() const;
+        Vec3          getPreRotation() const;
+        Vec3          getLocalTranslation() const;
+        Vec3          getLocalRotation() const;
+        Vec3          getLocalScaling() const;
+        Matrix        getGlobalTransform() const;
+        Matrix        getLocalTransform() const;
+        Matrix        evalLocal(const Vec3& translation, const Vec3& rotation) const;
+        Matrix        evalLocal(const Vec3& translation, const Vec3& rotation, const Vec3& scaling) const;
+        bool          isNode() const { return is_node; }
+
+        template <typename T>
+        T* resolveObjectLink(int idx) const
+        {
+            return static_cast<T*>(resolveObjectLink(T::s_type, nullptr, idx));
+        }
+
+        u64             id;
+        char            name[128];
+        const IElement& element;
+        const Object*   node_attribute;
+
+    protected:
+        bool         is_node;
+        const Scene& scene;
+    };
+
+    struct Pose : Object
+    {
+        static const Type s_type = Type::POSE;
+        Pose(const Scene& _scene, const IElement& _element);
+
+        virtual Matrix        getMatrix() const = 0;
+        virtual const Object* getNode() const = 0;
+    };
+
+    struct Texture : Object
+    {
+        enum TextureType
+        {
+            DIFFUSE,
+            NORMAL,
+            SPECULAR,
+
+            COUNT
+        };
+
+        static const Type s_type = Type::TEXTURE;
+
+        Texture(const Scene& _scene, const IElement& _element);
+        virtual DataView getFileName() const = 0;
+        virtual DataView getRelativeFileName() const = 0;
+    };
+
+    struct Material : Object
+    {
+        static const Type s_type = Type::MATERIAL;
+
+        Material(const Scene& _scene, const IElement& _element);
+
+        virtual Color          getDiffuseColor() const = 0;
+        virtual Color          getSpecularColor() const = 0;
+        virtual const Texture* getTexture(Texture::TextureType type) const = 0;
+    };
+
+    struct Cluster : Object
+    {
+        static const Type s_type = Type::CLUSTER;
+
+        Cluster(const Scene& _scene, const IElement& _element);
+
+        virtual const int*    getIndices() const = 0;
+        virtual int           getIndicesCount() const = 0;
```

Comment:
`getPlurals()` and `getSingularCount()`. So `getIndexCount` in the original code is correct.